diff.py

```diff
@@ -2425,10 +2451,21 @@ def do_diff(lines1: List[Line], lines2: List[Line], config: Config) -> Diff:
                 out1 += Text(address1, address_imm_fmt)
                 out2 += Text(address2, address_imm_fmt)
         elif line1 and line2:
-            line_prefix = "|"
-            line_color1 = line_color2 = sym_color = BasicFormat.DIFF_CHANGE
-            out1 = out1.reformat(line_color1)
-            out2 = out2.reformat(line_color2)
+            line1_parts = [line1.mnemonic] + line1.original.split("\t")[1].split(",")
+            line2_parts = [line2.mnemonic] + line2.original.split("\t")[1].split(",")
+            ### check each part of the line for innequality, but ignore parts with match everything symbol
+            for i in range(len(line1_parts)):
+                # If line2 (right side) is a match to any symbol case
+                if imm_matches_everything(line2_parts[i], config.arch):
+                    # And line1 (left side) had a relocation, then ignore this mismatch
+                    if line1.relocation_occured:
+                        continue
+                if line1_parts[i] != line2_parts[i]:
+                    line_prefix = "|"
+                    line_color1 = line_color2 = sym_color = BasicFormat.DIFF_CHANGE
+                    out1 = out1.reformat(line_color1)
+                    out2 = out2.reformat(line_color2)
+                    break
```

Comment:
Why is this needed? This is supposed to be handled by the `line1.diff_row == line2.diff_row` case

Reply:
messaged you on discord about this, think I got it resolved.

Reply:
nevermind, this is still needed, its that darn r2/0 r13/0 swaps.  I figured out some other workarounds to make this work in the  `line1.diff_row == line2.diff_row` case, however, I think you will not like those workarounds as much as what is here.